app/helpers/wifi_density_helper.rb

```diff
@@ -0,0 +1,94 @@
+module WifiDensityHelper
+  CACHE_DURATION = Rails.env.development? ? 10.seconds : 5.minutes
+
+  def wifi_density_percentage(location)
+    # If this location's code is not mapped in the wifi density config file,
+    # that means wifi density info is not supported for the location.
+    config_for_location = WIFI_DENSITY[location.code]
+    return nil if config_for_location.nil? || !config_for_location.key?('high') || !config_for_location.key?('cuit_location_ids')
+
+    Rails.cache.fetch('wifi_density_data-' + location.code, expires_in: CACHE_DURATION) do
+      wifi_density_data_for_locations = Rails.cache.fetch('wifi_density_data', expires_in: CACHE_DURATION) do
+        fetch_hierarchical_wifi_density_data
+      end
+      aggregated_locations = {}
+
+      config_for_location['cuit_location_ids'].each do |cuit_location_id|
+        location_data = wifi_density_data_for_locations[cuit_location_id.to_s]
+        next if location_data.nil?
+        aggregated_locations.merge!({cuit_location_id.to_s => location_data})
+        aggregated_locations.merge!(recursively_collect_children(location_data))
+      end
+
+      # Sum client_count totals from all aggregated locations
+      total_client_count = 0
+      aggregated_locations.each do |location_id, location_data|
+        total_client_count += location_data.fetch('client_count', 0).to_i
+      end
+
+      # If total client count is actually reported to be zero, return zero
+      return 0 if total_client_count == 0
+
+      # Make sure to use float division here so we can calculate a percentage.
+      percentage_integer = ((total_client_count / config_for_location['high'].to_f) * 100).round
+      if percentage_integer < 1
+        # If there is at least one client, it seems incorrect to say 0%, so always return a minimum of 1%.
+        1
+      elsif percentage_integer > 100
+        # If the expected high was set too low (in wifi_density.yml), we may get a percentage above 100%, so cap at 100%.
+        100
+      else
+        percentage_integer
+      end
+    end
+  end
+
+  def fetch_raw_wifi_density_data
+    if Rails.env.development? || Rails.env.test?
+      JSON.parse(File.read(Rails.root.join('spec/fixtures/files/sample-wifi-density-response.json')))
+    else
+      JSON.parse(Net::HTTP.get(URI(root_url + "wifi-density.json")))
+    end
+  end
```

Comment:
I wonder if a configuration setting here would be better than just having the file paths listed in this code, but its a little tricky because one is a url request and one is not.

Reply:
I thought a little more about what I would do here. For the testing environment, I would use WebMock to mock the request for the wifi-density. That way you can also test that the response is being made the way you expected. Though this solution doesn't provide an alternative for development. Just a thought.

Reply:
We should find a way to configure this.